### PR TITLE
Add ability to have linked images in figures

### PR DESCRIPTION
**Pull request by mayo**
Add support for [](http://link) and treat it the same way as standalone image.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -16,6 +16,13 @@ describe('markdown-it-implicit-figures', function() {
     assert.equal(res, expected);
   });
 
+  it('should add <figure> when image is by itself in a paragraph or preceeded by a standalone link', function () {
+    md = Md().use(implicitFigures, { dataType: true, figcaption: true });
+    var src = 'text with ![](img.png)\n\n![](fig.png)\n\nanother paragraph\n\n[![Caption](fig.png)](http://example.com)\n\nyet another paragraph';
```
Comment (arve0) [inline test.js:21]: Can you shorten this to be just the `[](url.html)`?

**Comment by arve0** [comment]
Thanks! Released in v0.3.0

File: test.js (hunk 1 above)
Comment (mayo) [inline test.js:21]: I could shorten the url link (to url.html), but I inserted the caption as the code that places figcaption was modified, so wanted to make sure it places the tags in the right spot. I guess it's too late now that you merged :)

Thanks for the quick merge!

Comment (arve0) [inline test.js:21]: Good point. I'll fix that.

Comment (arve0) [inline test.js:21]: Hm, I did not remove the [figcaption](https://github.com/arve0/markdown-it-implicit-figures/blob/master/test.js#L21)?
